src/cpp/rtps/RTPSDomain.cpp

```diff
@@ -214,6 +214,7 @@ bool RTPSDomain::removeRTPSParticipant(
 {
     if (p != nullptr)
     {
+        assert((p->mp_impl != nullptr) && "This participant has been previously invalidated");
```

Comment:
Shouldn't we then add `mp_impl = nullptr` to the destructor of `RTPSParticipant`